sigstore-java/src/main/java/dev/sigstore/bundle/BundleFactory.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.bundle;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import dev.sigstore.KeylessSigningResult;
+import dev.sigstore.proto.bundle.v1.Bundle;
+import dev.sigstore.proto.bundle.v1.VerificationMaterial;
+import dev.sigstore.proto.common.v1.*;
+import dev.sigstore.proto.rekor.v1.*;
+import dev.sigstore.rekor.client.RekorEntry;
+import java.security.cert.CertificateEncodingException;
+import java.util.Base64;
+import java.util.stream.Collectors;
+
+/** Generates Sigstore Bundle. */
+public class BundleFactory {
```

Comment:
I wonder if we plan on using bundle everywhere, this could be exposed as an [`@Value.Derived`](https://immutables.github.io/immutable.html#derived-attributes) member on the KeylessSigningResult type.

Reply:
The trick is that Bundle can include RFC3161 timestamp which does not derive from KeylessSigningResult

Reply:
On the other hand, we can expose `Bundle` (proto object) as `@Derived`, then RFC3161 can be added later.
However, adding `Bundle` to `KeylessSigningResult` would make `dev.sigstore:protobuf-specs` and `com.google.protobuf:protobuf-java-util` mandatory dependencies :-/

If we keep `Bundle` away from `KeylessSigningResult`, then we could place them to separate artifacts with different set of dependencies.